### DISCUSSION TITLE
Kafka Consumer Activity

### DISCUSSION THREAD

**Discussion by tulrichtrimble** [elsa-workflows/elsa-core#5175] [Design & tech]
There was a [Kafka Consumer](https://hexdocs.pm/elsa_kafka/Elsa.Consumer.html) in Elsa 2.0. Are there any plans, community or otherwise, to implement a Kafka Consumer activity in 3.0?

If not, could anyone offer any direction before we invest too much time reverse engineering the code? For example, are there scalability considerations to pausing hundreds of workflow instances while they await a message? The [docs on Triggers](https://v3.elsaworkflows.io/docs/core-concepts/trigger) seem to be a work in progress.

Thanks!

**Comment by glime-ai** [comment]
As of my knowledge cutoff date, there is no direct mention in the provided context of a Kafka Consumer activity specifically for Elsa 3.0. However, considering the open-source nature of Elsa and the community contributions, it is possible that a Kafka Consumer activity could be implemented by the community or by the Elsa team in the future.

Regarding the direction before investing time in reverse engineering the code, it would be advisable to review the existing Elsa 3.0 documentation, explore the GitHub repository for any ongoing discussions or pull requests related to Kafka, and consider reaching out to the community through their Discord channel or Stack Overflow for insights and guidance.

Scalability considerations for pausing hundreds of workflow instances while they await a message would depend on the underlying infrastructure and how Elsa is configured. Elsa 3.0 introduces a distributed runtime using Proto.Actor, which could potentially help with scalability. However, specific details on handling large numbers of paused workflow instances would likely require a deeper dive into the Elsa 3.0 architecture and possibly custom solutions.

For the most accurate and up-to-date information, I recommend checking the official Elsa documentation, GitHub issues, and discussions, as well as engaging with the Elsa community.

**Reply by douglasg14b** [reply]
Is the `master` branch of this repo Elsa 2.0? I see PRs for that branch that touch the Kafka activity 🤔
